src/theory/quantifiers/instantiate.cpp

```diff
@@ -591,19 +591,13 @@ bool Instantiate::recordInstantiationInternal(Node q, std::vector<Node>& terms)
   {
     Trace("inst-add-debug")
         << "Adding into context-dependent inst trie" << std::endl;
-    CDInstMatchTrie* imt;
-    std::map<Node, CDInstMatchTrie*>::iterator it = d_c_inst_match_trie.find(q);
-    if (it != d_c_inst_match_trie.end())
-    {
-      imt = it->second;
-    }
-    else
+    const auto [it, wasInserted] = d_c_inst_match_trie.insert({q, nullptr});
```

Comment:
I'd prefer not using this syntax, can we just use `.first` / `.second` as needed?

Reply:
I would say that this finally gives you a way of returning multiple values in C++,  so we might be missing out. Should this may be part of the guidelines? (The use of modern C++17 features in general)

Reply:
We had a lengthy discussion internally about this point in particular. We decided that these "very new" features can be used, but are not mandated, and every developer can decide whether he wants to have/use them in the code he feels responsible for.
As for this PR, @ajreynol feels responsible for this part of the code and thus there will be no structured bindings for the time being :-).

Reply:
Structured binding removed from this request.